linalg/ode.hpp

```diff
@@ -931,6 +938,100 @@ class WBZAlphaSolver : public GeneralizedAlpha2Solver
 
 };
 
+/// Class for solving systems of split ODEs:
+/// M dx/dt = f_1(x,t) + f_2(x,t)
+/// where M^{-1}*f_1 and M^{-1}*f_2 are treated differently (e.g.,
+/// explicitly and implicitly).This is the abstract base class for
+/// such ODE solvers expecting a TimeDependentOperator with split
+/// functionality. Setting EvalMode=ADDITIVE_TERM_1 and calling
+/// Mult(...) should return k1=M^{-1}*f_1(x,t). Setting
+/// EvalMode=ADDITIVE_TERM_2 and calling ImplicitSolve(...) should
+/// solve M*k2 = f_2(x+gamma*k2,t).
+class SplitODESolver : public ODESolver
```

Comment:
I'm not sure we need this class -- the `Select` method can be added to `ODESolver` as `SelectIMEX`.

Maybe one reason to keep this class is that it provides a single place where the requirements that all IMEX ODE solvers (derived from this class) have for the `TimeDependentOperator` are explicitly written down -- it needs to implement `Mult` and `ImplicitSolve` with support for setting the eval-mode with `SetEvalMode`. The minimum requirement may actually be to have `Mult` with support for `EvalMode == ADDITIVE_TERM_1` (explicit part) and `ImplicitSolve` with support for `EvalMode == ADDITIVE_TERM_2` (implicit part). This documentation is already written, so it's good to keep it in some form.

Reply:
I'm also in favor of not introducing `SplitODESolver` (and instead adding `ODESolver::SelectIMEX`).  Unless `SplitODESolver` is adding functionality over `ODESolver`, I personally prefer to not add another level of inheritance.

I'll note that "split" is ambiguous in a way that can be problematic if other splittings / partitionings are considered (e.g. nonlinear partitionings).  On the time integrator side of things, the equations this targets are specifically "additively split" or "additively partitioned"

Reply:
I will note that having all the IMEX solvers inherit directly from `ODESolver` instead of `SplitODESolver` will make inserting a common base class along the lines of `ARKSolver` suggested by @sohailreddy .  Having `SplitODESolver` introduced now as-is opens the door to someone else deriving `SplitODESolver` in a way that makes introducing `ARKSolver` in a future PR more difficult.

Reply:
@cjvogl, if `SplitODESolver` is removed, where do you think is the best place to move its documentation? I think we should keep that in some form.

Reply:
> @cjvogl, if SplitODESolver is removed, where do you think is the best place to move its documentation? I think we should keep that in some form.

Because the documentation describes how things defined in `TimeDependentOperator` are used, that is where I personally would put it ... but I think `ODESolver` works too with some advantages.